packages/frontend/src/components/common/Filters/FilterRuleForm.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button, Colors, HTMLSelect } from '@blueprintjs/core';
+import {
+    DimensionType,
+    fieldId as getFieldId,
+    FilterableDimension,
+    FilterOperator,
+    FilterRule,
+} from 'common';
+import React, { FC, useCallback, useMemo } from 'react';
+import { FilterTypeConfig } from './configs';
+import FieldAutoComplete from './FieldAutoComplete';
+
+type Props = {
+    fields: FilterableDimension[];
+    filterRule: FilterRule;
+    onChange: (value: FilterRule) => void;
+    onDelete: () => void;
+};
+
+const FilterRuleForm: FC<Props> = ({
+    fields,
+    filterRule,
+    onChange,
+    onDelete,
+}) => {
+    const activeField = fields.find(
+        (field) => getFieldId(field) === filterRule.target.fieldId,
+    );
+
+    const filterConfig = useMemo(
+        () => FilterTypeConfig[activeField?.type || DimensionType.STRING],
+        [activeField],
+    );
+
+    const onFieldChange = useCallback(
+        (fieldId: string) => {
+            const selectedField = fields.find(
+                (field) => getFieldId(field) === fieldId,
+            );
+            if (selectedField && activeField) {
+                if (selectedField.type === activeField.type) {
+                    onChange({
+                        ...filterRule,
+                        target: {
+                            fieldId,
+                        },
+                    });
+                } else {
+                    onChange({
+                        ...filterRule,
+                        target: {
+                            fieldId,
+                        },
+                        operator: FilterOperator.NULL,
+                        settings: undefined,
+                        values: undefined,
+                    });
+                }
+            }
+        },
+        [activeField, fields, filterRule, onChange],
+    );
+
+    return (
```

Comment:
Is this intentionally full width? 

<img width="764" alt="Screenshot 2022-01-31 at 16 00 46" src="https://user-images.githubusercontent.com/11660098/151828294-166dea63-c3a8-45bf-bbc8-9299291b7c77.png">

Reply:
Yes, so all rows independently of the input feel the same.